Tests cases specification

pietercolpaert opened the pull request:
Working on a proposal to add tests cases in the spec for #97

pietercolpaert committed:
Added first test cases for init

pietercolpaert committed:
Correct title

pietercolpaert committed:
Minor fixes

pietercolpaert committed:
Test cases further worked out

pietercolpaert committed:
Remove chapter numbers

smessie commented on 04-tests.bs:
```diff
@@ -0,0 +1,128 @@
+
+Purpose: Ensuring that it is unambiguous yet user friendly way of knowing the identifier of the collection, as well as the current page of the collection’s view.
+
+This test is about a function `initialization(url)`. The function will return a the collection and the URL of the root node.
```
```suggestion
This test is about a function `initialization(url)`. The function will return the URL of the collection and the URL of the root node.
```

smessie commented on 04-tests.bs:
```diff
@@ -0,0 +1,128 @@
+
+We’ll test this with various URLs containing different contents.
+
+### The URL to a rootnode without redirects ### {#1.1}
```
I feel like it could improve clarity to describe more extensively what the test should be testing. One way we could do that is by leveraging the `test` or `it` keywords commonly used by test frameworks: 
```suggestion
### Test if initializing with the URL to a rootnode without redirects returns the Collection URL and Root node URL ### {#1.1}
```
or 
```suggestion
### It should return the Collection URL and Root node URL when initialising with the URL to a rootnode without redirects ### {#1.1}
```

However we might not like it that the subtitles become very long, so an alternative idea could be to always include such a clear description right under the subtitle:
```suggestion
### The URL to a rootnode without redirects ### {#1.1}

**Test:** Initializing with the URL to a rootnode without redirects returns the Collection URL and Root node URL
```

I think this might make things clearer for people that want to implement the test suite.

smessie commented on 04-tests.bs:
```diff
@@ -0,0 +1,128 @@
+
+### The URL to a rootnode without redirects ### {#1.1}
+
+File url `test1.1.ttl`
```
I would consistently go for the uppercase URL throughout the spec
```suggestion
File URL `test1.1.ttl`
```

smessie commented on 04-tests.bs:
````diff
@@ -0,0 +1,128 @@
+
+```turtle
+@prefix tree: <https://w3id.org/tree#> .
+<test1.2.ttl#collection> tree:view <test1.2.ttl> .
````
I don't think it is necessarily wrong, but wouldn't it be an idea to improve clarity by always using the uppercased`<testX.X.ttl#Collection>` for a Collection, instead of once `<Collection>` and the other time `<test1.2.ttl#collection>`. Similarly we could opt for `<testX.X.ttl#Node>` as Node URL, then we also test that it correctly returns the URL to the Node, and not just the URL to the RDF Document (as I think this is the intended behaviour).

pietercolpaert replied:
Sure!

pietercolpaert replied:
Solved this

smessie reviewed (commented):
Just leaving some thoughts here while going through the current draft.

pietercolpaert committed:
Update 04-tests.bs

Co-authored-by: Ieben Smessaert <smessie@users.noreply.github.com>

pietercolpaert committed:
Update 04-tests.bs

Co-authored-by: Ieben Smessaert <smessie@users.noreply.github.com>

pietercolpaert committed:
Apply suggestions from code review

Co-authored-by: Ieben Smessaert <smessie@users.noreply.github.com>

pietercolpaert committed:
Further tests based on code review

xdxxxdx commented on 04-tests.bs:
```diff
@@ -0,0 +1,155 @@
+
+This test is about a function `initialization(url)`. The function will return the URL of the collection and the URL of the root node.
+
+We will test this with various URLs containing different contents.
```
It might be better to explain the purpose of the TREE test specification (perhaps in an overview section). For example (correct me if I am wrong):

> The TREE test specification aims to provide a set of tests for validating that a tree client implementation conforms to the TREE specification. The TREE implementors can use this conforming client to interact with data sources to validate TREE conformity.


Additionally, it might be clearer (to people like me XD) to explain what the initialization function does. For example:

> The `initialization(url)` function returns the URL of the collection and the URL of the root node.
> 
> The following test scenarios describe how the `initialization(url)` function of the tree client should behave in various situations involving the TREE and the parameter URL.

xdxxxdx committed:
feat: update test specification

- Restructured the documentation to create a narrative.
- Addressed my concern regarding the comment left in [issue #141](https://github.com/TREEcg/specification/issues/141), questioning whether this specification is ONLY focused on client behavior.
- OFC, a work in progress

xdxxxdx commented:
Hi @pietercolpaert,  
Just to let you know, I don't have write access to this branch. I've created another PR [#142](https://github.com/TREEcg/specification/pull/142) to continue making progress. (it is fine, I'll proceed this way.)

pietercolpaert committed:
Merge pull request #142 from xdxxxdx/feat-tests

feat: update test specification

xdxxxdx commented:
https://github.com/rorlic/tree-shacl/issues/1
- https://dl.acm.org/doi/abs/10.1145/3696410.3714694
-----
- I envision that the spec tests can only test a tree:Node (basic elements of the Tree), or a part of the full tree graph, therefore I made these two categories in the conformance points.
- The tests, in my view, need to emphasize the minimum requirements of a TREE implementation (the content of the TREE).
- I included the generated HTML(and will remove the HTML once the PR is settled) in the PR as well. Please feel free to comment on the current version. I would like to collect feedback on the current format.

xdxxxdx committed:
Update 04-tests.bs

xdxxxdx committed:
Update 04-tests.bs

xdxxxdx committed:
Update 04-tests.bs

xdxxxdx committed:
feat: add test spec

- I envision that the spec tests can only test a tree:Node (basic elements of the Tree), or a part of the full tree graph, therefore I made these two categories in the conformance points.
- The tests, in my view, need to emphasize the minimum requirements of a TREE implementation (the content of the TREE).
- I included the generated HTML in the PR as well. Please feel free to comment on the current version. I would like to collect feedback on the current format and will remove the HTML once the PR is settled.

xdxxxdx committed:
Update 04-tests.bs

xdxxxdx committed:
feat: delete redundancy

xdxxxdx committed:
Update 04-tests.bs

xdxxxdx committed:
Merge pull request #153 from TREEcg/master

fix; merge the update of the main branch

xdxxxdx committed:
Delete 04-tests.html

pietercolpaert commented:
Instead we’re going for #154
